src/hooks/useAccounts.tsx

```diff
@@ -224,7 +243,17 @@ const useAccountsContext = () => {
 
           const evmDerivedAccount = evmDerivedAddresses[evmAddress];
 
-          if (evmDerivedAccount?.encryptedSignature) {
+          if (walletConnectionType === WalletConnectionType.Privy && authenticated && ready) {
+            try {
+              const signature = await signTypedDataAsync();
```

Comment:
If the walletConnectionType is Privy, we will get the signature and use the signature to create the dydxWallet via `setWalletFromEvmSignature`.